src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx

```diff
@@ -382,12 +331,20 @@ function BarplotWithControls({
   onDependentAxisLogScaleChange,
   valueSpec,
   onValueSpecChange,
+  updateThumbnail,
   ...barPlotProps
 }: BarplotWithControlsProps) {
+  const ref = useRef<PlotRef>(null);
+  useEffect(() => {
+    ref.current
+      ?.toImage({ format: 'svg', ...plotDimensions })
+      .then((src) => updateThumbnail(src));
+  }, [data, updateThumbnail]);
```

Comment:
Non-back-end controls need to be passed too
```suggestion
  }, [data, updateThumbnail, dependentAxisLogScale]);
```